packages/grafana-data/src/panel/PanelPlugin.ts

```diff
@@ -42,6 +63,81 @@ export class PanelPlugin<TOptions = any> extends GrafanaPlugin<PanelPluginMeta>
     this.panel = panel;
   }
 
+  get defaults() {
+    let result = this._defaults || {};
+
+    if (!this._defaults) {
+      const editors = this.optionEditors;
+
+      if (!editors || editors.list().length === 0) {
+        return null;
+      }
+
+      for (const editor of editors.list()) {
+        set(result, editor.id, editor.defaultValue);
+      }
+    }
+    return result;
+  }
+
+  get fieldConfigDefaults(): FieldConfigSource<TFieldConfigOptions> {
+    let customPropertiesDefaults = this._fieldConfigDefaults.defaults.custom;
+
+    if (!customPropertiesDefaults) {
+      customPropertiesDefaults = {} as TFieldConfigOptions;
+    }
+    const editors = this.customFieldConfigs;
+
+    if (editors && editors.list().length !== 0) {
+      for (const editor of editors.list()) {
+        set(customPropertiesDefaults, editor.id, editor.defaultValue);
+      }
+    }
+
+    return {
+      defaults: {
+        ...(this._standardFieldConfigProperties ? Object.fromEntries(this._standardFieldConfigProperties) : {}),
+        custom:
+          Object.keys(customPropertiesDefaults).length > 0
+            ? {
+                ...customPropertiesDefaults,
+              }
+            : undefined,
+        ...this._fieldConfigDefaults.defaults,
+      },
+      // TODO: not sure yet what about overrides, if anything
+      overrides: this._fieldConfigDefaults.overrides,
```

Comment:
@ryantxu maybe you have some thoughts on what should we do about default overrides? Personally I don't see why default overrides should be set.

Reply:
I think we can wait with that for now